concrete-core/src/backends/core/private/crypto/glwe/fourier.rs

```diff
@@ -0,0 +1,308 @@
+    pub fn fill_with_forward_fourier<InputCont>(
+        &mut self,
+        glwe: &GlweCiphertext<InputCont>,
+        buffers: &mut FftBuffers,
+    ) where
+        Cont: AsMutSlice<Element = Complex64>,
+        GlweCiphertext<InputCont>: AsRefTensor<Element = Scalar>,
+        Scalar: UnsignedTorus,
+    {
+        // We retrieve a buffer for the fft.
+        let fft_buffer = &mut buffers.first_buffer;
+        let fft = &mut buffers.fft;
+
+        // We move every polynomial to the fourier domain.
+        let poly_list = glwe.as_polynomial_list();
+        let iterator = self
+            .tensor
+            .subtensor_iter_mut(self.poly_size.0)
+            .map(|t| FourierPolynomial::from_container(t.into_container()))
+            .zip(poly_list.polynomial_iter());
+        for (mut fourier_poly, coef_poly) in iterator {
+            fft.forward_as_torus(fft_buffer, &coef_poly);
+            fourier_poly
+                .as_mut_tensor()
+                .fill_with_one((fft_buffer).as_tensor(), |a| *a);
+        }
+    }
+
+    /// Fills a GLWE ciphertext with the inverse fourier transform of a Fourier GLWE ciphertext
+    /// ```
+    /// use crate::backends::core::private::crypto::bootstrap::fourier::buffers::FftBuffers;
+    /// use concrete_commons::parameters::{GlweSize, PolynomialSize};
+    /// use concrete_core::backends::core::private::crypto::glwe::FourierGlweCiphertext;
+    /// use concrete_core::backends::core::private::math::fft::Complex64;
+    ///
+    /// let mut fourier_glwe: FourierGlweCiphertext<_, u32> =
+    ///     FourierGlweCiphertext::allocate(Complex64::new(0., 0.), PolynomialSize(10), GlweSize(7));
+    /// let buffers_in = &mut buffers.first_buffer;
+    /// let buffers_out = &mut buffers.second_buffer;
+    /// let glwe_in = GlweCiphertext::allocate(10 as u8, PolynomialSize(10), GlweSize(7));
+    ///
+    /// fourier_glwe.fill_with_forward_fourier(glwe_in, buffers_in);
+    ///
+    /// let glwe_out = GlweCiphertext::allocate(0 as u8, PolynomialSize(10), GlweSize(7));
+    ///
+    /// fourier_glwe.fill_with_backward_fourier(glwe_out, buffers_out);
+    /// ```
+    pub fn fill_with_backward_fourier<InputCont, Scalar_>(
+        &mut self,
+        glwe: &mut GlweCiphertext<InputCont>,
+        buffers: &mut FftBuffers,
+    ) where
+        Cont: AsMutSlice<Element = Complex64>,
+        GlweCiphertext<InputCont>: AsMutTensor<Element = Scalar_>,
+        Scalar_: UnsignedTorus,
+    {
+        // We retrieve a buffer for the fft.
+        let fft = &mut buffers.fft;
+
+        // we move every polynomial to the coefficient domain
+        let iterator = glwe
+            .as_mut_tensor()
+            .subtensor_iter_mut(self.poly_size.0)
+            .map(|t| Polynomial::from_container(t.into_container()))
+            .zip(self.polynomial_iter_mut());
+
+        for (mut coef_poly, mut fourier_poly) in iterator {
+            fft.add_backward_as_torus(&mut coef_poly, &mut fourier_poly);
+        }
```

Comment:
Here, I think we should add an entry point `backward_as_torus` which oonly performs the backward transform (not the addition).

Reply:
Had a go at doing this, also added `backward_two_as_torus`